### PR TITLE
Implement delete operations

### PR DESCRIPTION
**Pull request by artur-ciocanu**

### CODE REVIEW

File: dapr-spring-core/src/main/java/io/diagrid/spring/core/keyvalue/DaprKeyValueAdapter.java [hunk 1]
```diff
@@ -122,13 +124,18 @@ public CloseableIterator<Entry<Object, Object>> entries(String keyspace) {
...
+        var keyspaceFilter = getKeyspaceFilter(keyspace);
+        var sql = String.format(DELETE_BY_KEYSPACE_SQL_PATTERN, keyspaceFilter);
+        var meta = Map.of("sql", sql);
+
+        daprClient.invokeBinding(stateStoreBinding, "exec", null, meta).block();
     }
 
     @Override
     public void clear() {
+        var meta = Map.of("sql", DELETE_ALL_SQL);
```
Comment (salaboy) [inline dapr-spring-core/src/main/java/io/diagrid/spring/core/keyvalue/DaprKeyValueAdapter.java:136]: Is this necessary? Does this needs to clean up all the keys?

File: dapr-spring-core/src/test/java/io/diagrid/spring/core/kvstore/DaprKeyValueTemplateIT.java [hunk 2]
```diff
@@ -103,4 +103,30 @@ public void testUpdateDaprKeyValueTemplate() {
         assertThat(updatedType).isNotNull();
     }
 
+    @Test
+    public void testDestroyDaprKeyValueTemplate() throws Exception {
+        var itemId = 1;
+        var insertedType = keyValueTemplate.insert(new TestType(itemId, "test"));
+        assertThat(insertedType).isNotNull();
+
+        keyValueTemplate.destroy();
```
Comment (salaboy) [inline dapr-spring-core/src/test/java/io/diagrid/spring/core/kvstore/DaprKeyValueTemplateIT.java:112]: We need to check other implementations to make sure this is the right behaviour?

File: dapr-spring-core/src/main/java/io/diagrid/spring/core/keyvalue/DaprKeyValueAdapter.java (hunk 1 above)
Comment (artur-ciocanu) [reply dapr-spring-core/src/main/java/io/diagrid/spring/core/keyvalue/DaprKeyValueAdapter.java:136]: If you think it is too dangerous we can leave it as `Unsupported....`. I took inspiration from `MapKeyValueAdapter`

File: dapr-spring-core/src/test/java/io/diagrid/spring/core/kvstore/DaprKeyValueTemplateIT.java (hunk 2 above)
Comment (artur-ciocanu) [reply dapr-spring-core/src/test/java/io/diagrid/spring/core/kvstore/DaprKeyValueTemplateIT.java:112]: In `Spring Data KeyValue` `destroy()` clears the underlying map. Similarly to previous comment we can add `UnsupportedOperationException` so it is clear that we don't support this destructive operation.

File: dapr-spring-core/src/main/java/io/diagrid/spring/core/keyvalue/DaprKeyValueAdapter.java (hunk 1 above)
Comment (salaboy) [reply dapr-spring-core/src/main/java/io/diagrid/spring/core/keyvalue/DaprKeyValueAdapter.java:136]: Yeah I wouldn't follow the in memory implementation but the Redis one..

**Review by salaboy — APPROVED** [review]

**Pull request merged**